src/code/z_debug.c

```diff
@@ -282,14 +281,14 @@ void func_80063D7C(GraphicsContext* gfxCtx) {
     GfxPrint_Open(&printer, gfx);
 
     if ((OREG(0) == 1) || (OREG(0) == 8)) {
-        func_80063828(&printer);
+        Debug_DrawOnScreenText(&printer);
```

Comment:
Important to note that this text will only draw if these O(gawa) regs are set
This is only used in camera, debug cam, and that one call for noclip in player. Its prob safe to say that the camera dev implemented noclip

That said, I opted to name it more generally and not after camera. Wasnt totally sure what to do, but I think general is slightly preferable when its only tie to camera is the regs that allow it to draw

Reply:
This system does seem designed to be more generalized with one exception: these `OREG`s are annoying because if an external system changes `OREG(0)` to draw the screen text, then `Camera_Init` on the main camera (i.e. upon `Play_Init`) will overwrite `OREG(0)` based on `sOREGInit`:

```c
for (i = 0; i < sOREGInitCnt; i++) {
    OREG(i) = sOREGInit[i];
}
```

Reply:
yea
my main reasoning for keeping the name general is because the text functions themselves arent really tied in any way. only the call to draw it in the `Debug_Draw` function (which we are still discussing

Reply:
Kinda brainstorming but what about naming this "screen debug text" system after the O in OREG (or non-canonically O for Ogawa)
like:
- `Debug_ScreenTextColored` -> `Debug_OTextColored`
- `Debug_DrawOnScreenText` -> `Debug_DrawOText`
- `DebugTextBufferEntry sDebugTextBuffer` -> `DebugOTextBufferEntry sDebugOTextBuffer`

:sunglasses:

Reply:
yea I still think I want to not tie it to Oreg for now, since nothing about the functions are actually tied to it, just how its called in debug_draw
But as you mention in the comment down below, we may have a better naming scheme for these debug related things overall later. Opinion may change then